lib/Net/SAML2/SP.pm

```diff
@@ -343,7 +345,12 @@ sub metadata {
                             $ds,
                             $self->_cert_text,
                         )
-                    )
+                    ),
+                    $x->KeyName(
+                     $ds,
+                     Digest::MD5::md5_hex($self->_cert_text)
```

Comment:
Why md5 hex?

Reply:
Because it generates a stable "name" for any cert without it being too long or what not.